### DISCUSSION TITLE
Resolving Label Overlap Issues in YOLO

### DISCUSSION THREAD

**Discussion by vdt104** [ultralytics/ultralytics#17382] [Q&A]
My model, trained with YOLO, predicts on the test set, and some vehicles labeled as cars (label 1) are being detected as both 0 (0 means motorbikes) and 1, with both bounding boxes almost overlapping. Ideally, those objects should only have label 1. What should I do to resolve this issue? I’ve tried using NMS, but it significantly reduces the evaluation mAP@50 score. 
![image](https://github.com/user-attachments/assets/cb4c2de9-efee-4ffb-8b3a-655e61e824fe)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @vdt104, thank you for reaching out and contributing to the Ultralytics community 🚀! We understand overlapping label predictions can be challenging.

We recommend taking a look at our comprehensive [Docs](https://docs.ultralytics.com) where you can find insights on using [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) with YOLO. Often, these resources can offer solutions to common issues.

Since you mentioned encountering a problem with NMS affecting your evaluation scores, understanding the context in which NMS is applied can be vital. If this is a 🐛 Bug Report, please help us assist you better by providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). This will enable us to dive deeper into potential issues.

Additionally, if this counts as a custom training ❓ Question, providing detailed information, including dataset examples and training logs, is crucial for us to give precise advice. Make sure also to follow our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

You're also welcome to engage with the Ultralytics community for more support. For real-time conversations, join us on [Discord](https://ultralytics.com/discord) 🎧. For more detailed discussions, visit our [Discourse](https://community.ultralytics.com), or explore various threads on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

To ensure any recent patches or improvements potentially relevant to your issue have been applied, upgrade to the latest `ultralytics` package along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

Consider trying YOLO in these verified environments, each preconfigured with necessary dependencies including [CUDA](https://developer.nvidia.com/cuda), [CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. Explore our [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

This badge reflects the current status of our [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests. If green, all tests validating YOLO operations on macOS, Windows, and Ubuntu are passing, confirming current stability.

This response is automated, but an Ultralytics engineer will review your discussion and provide additional assistance soon. 😊

**Comment by glenn-jocher** [comment]
@vdt104 to address label overlap issues, consider adjusting the confidence threshold and IoU threshold for Non-Maximum Suppression (NMS) to better differentiate between classes. Additionally, reviewing and refining your dataset annotations might help improve model accuracy.
